spikeinterface/core/segmentutils.py

```diff
@@ -218,6 +221,63 @@ def concatenate_recordings(*args, **kwargs):
 concatenate_recordings.__doc__ == ConcatenateSegmentRecording.__doc__
 
 
+class SelectSegmentRecording(BaseRecording):
+    """
+    Return a new recording with a single segment from a multi-segment recording.
+
+    Parameters
+    ----------
+    recording : BaseRecording
+        The multi-segment recording
+    segment_indices : list of int
+        The segment indices to select
+    """
+
+    def __init__(self, recording: BaseRecording, segment_indices: List[int]):
+        BaseRecording.__init__(self, recording.get_sampling_frequency(), 
+                               recording.channel_ids, recording.get_dtype())
+        recording.copy_metadata(self)
+        
+        num_segments = recording.get_num_segments()
+        assert len(segment_indices) < num_segments, f"'segment_index' must be between 0 and {num_segments - 1}"
+
+        for segment_index in segment_indices:
+            rec_seg = recording._recording_segments[segment_index]
+            self.add_recording_segment(rec_seg)
+
+        self._kwargs = {'recording': recording.to_dict(),
+                        'segment_indices': segment_indices}
```

Comment:
You neeed to ensure that segment_indices is not a numpy array
segment_indices = segment_indices[int(e) for a in segment_indices]